packages/flutter/lib/src/widgets/router.dart

```diff
@@ -899,9 +899,9 @@ abstract class BackButtonDispatcher extends _CallbackHookProvider<Future<bool>>
   /// additionally attempt to claim priority from its parent, whereas removing
   /// the last child does not.)
   void forget(ChildBackButtonDispatcher child) {
-    assert(_children != null);
-    assert(_children!.contains(child));
-    _children!.remove(child);
+    if(_children != null) {
```

Comment:
Either this, or the child need to keep a state whether it has called take priority or not

Reply:
This is for the first improvement

Reply:
The first improvement being "ChildBackButtonDispatcher can be disposed without calling take priority"?

I don't fully understand what you mean by this improvement since ChildBackButtonDispatcher doesn't have a dispose method and hence doesn't need to be disposed?

Reply:
sorry for the confusing. When Router gets a new backbuttondispatcher in didUpdateWidget, it will call the removeCallback method on the old backbuttondispatcher which then will call parent.forget if there is no callback left

Reply:
Ah, makes sense. I wonder if we should just declare _children as `late final` and initialize it right there in the declaration. That way, It would also be lazily initialized like we want it right now.